File: other_problems/5_closure_test.js
var fibonacci = function (i) {
  console.log("fibonacci")
  memo = [0, 1, 2];

  var fib = function(i) {
    var result = memo[i];
    if(typeof result !== "number") {
      console.log("fib")
      result = fib(i-1) + fib(i-2);
      memo[i] = result;
    }
    return result;
  };

  return fib;
}();

for(var i = 0; i < 10; i++) {
  console.log(fibonacci(i))
}

File: other_problems/5_closure_test.rb
def fabonacci
  memo = [0, 1, 2]

  lambda = lambda do |i|
    result = memo[i]
    if result.nil?
      result = lambda.call(i-1) + lambda.call(i-2)
    end
    return result
  end

  return lambda
end

lambda = fabonacci

10.times do |i|
  p lambda.call(i)
end 
